### ISSUE TITLE
Highly probably most efficient workshops/buildings "matrix" placement?

### ISSUE THREAD

**Issue by Rakso69** [Sh0wdown/UnofficialCrusaderPatch#409]
Since ages SHC players always have been wondering about the best placement build-up strategies. I decided to playtest all of my ideas, others suggestions and new thoughts last month alot, and that's what my experiments have guided me to, and what I've accomplished in the very end:
![xDF0VaL](https://user-images.githubusercontent.com/45049934/59733110-05442580-924d-11e9-830b-4ea63712b2e1.png)
You're welcome to tryout using any sort of "little inches" of it in your own AI projects for highest efficiency production. Adjust it with your own preferences or suggest me changes on the placement to make it evenly even way better, till the limit! :)

**Comment by GRhin** [comment]
Nice work.
I assume your metrics are weapon production per minute (w/m)? Or is it gold value (g/m)?
Have you considered doing another version with a bread economy?

**Comment by GRhin** [comment]
Also have you considered deleting initial stockpile and building it somewhere else?
Also also for space efficiency, how much difference would it make to consolidate to make each row/column 2 buildings deep?

**Comment by Rakso69** [comment]
@GRhin Both, but bigger emphasis on gold.
Weapons would solely be just all poleturners without fletchers at all. I suggest still using only poleturners  for AI lords that make crossbowmen, because gold is prior to crossbows as it can be used onto different things at once which increases AI development speed rather than limit it to just a single source to a material and slow down growth.
Bread would look exactly same. In my own AIV project I've just changed a part of it to bread, and just kinda edited it whole. Here's a shot:
![unknown](https://user-images.githubusercontent.com/45049934/59752459-9255a180-9282-11e9-9796-9861bd67b706.png)
Deleting it and building somewhere else would only change the bottom left side(where the keep is) and nothing else. Of course that's a long-range most effective build-up placement for 4 and more stockpiles in usage. With 3, 2 or 1 it's being whole reshaped again, and the solution that you're describing with each row/column 2 buildings deep suggestion is my own solution for 3 and less stockpiles, that I do use on my Stronghold 1 Youtube videos, and nevertheless it was already been used on such a low ammount since ages.
[https://www.youtube.com/watch?v=LaiDZ4CgV6E&list=PLJXScksSEnht_wjsf-3rj9SRLjfMONUA_](My SH1 Vids)

**Comment by Krarilotus** [comment]
This is great!
I woud love to see you taking part in my upcoming AI development contest using the UCP.

I will announce on Saturday, It ll give pricemoney for sure, the more serious participants, the more prizemoney i ll put in. Lets figure out some more of these values and get AI to the next level!

**Comment by GRhin** [comment]
@Krarilotus this is probably not the best place to discuss your contest, but Id really like more info on how you are running it (not to participate, but as an observer), where can I get in contact with you about that?

**Comment by Krarilotus** [comment]
> @Krarilotus this is probably not the best place to discuss your contest, but Id really like more info on how you are running it (not to participate, but as an observer), where can I get in contact with you about that?

You can hit me with an email (jojo.krarilotus@gmail.com) or if its just related to the contest: shc.aicontest@gmail.com
Else you can write me on discord (nickname Krarilotus)

**Comment by GRhin** [comment]
whats the 4 digit code for discord? Or is there a server I should be on?

**Comment by Krarilotus** [comment]
The Server is named Udwins Heimatburg, you can go on that, if you know Udwins Youtube Channel, then click on the discord symbol on the right corner of his channel banner.

**Comment by Rakso69** [comment]
Update:
![Best Workshops Placement2](https://user-images.githubusercontent.com/45049934/60479609-bfdd1a80-9c86-11e9-81df-5627b838f328.png)

**Comment by Rakso69** [comment]
### **Update:**
![Best Workshops Placement2](https://user-images.githubusercontent.com/45049934/60480969-d89bff00-9c8b-11e9-95ee-7fec86eddf2f.png)

**Comment by GRhin** [comment]
why the change? from what I see the bottom half has been rearranged to be horizontally lined, as opposed to vertically, and the top center has some random gaps?

**Comment by Rakso69** [comment]
@GRhin All the changes you can notice save up from 2 to 4-5 less steps needed for the worker to travel.

If you don't understand changes you must have no clue about **door-change** rules in the game. Make a 2x2, 3x2 and 2x3 workshops boxes on singleplayer without armoury and see where the workers are, so you'll better understand my update. :) I edit the scheme by following these box rules.

I changed the whole scheme, because a friend suggested me some minor changes over which I have been experimenting around and which have guided me to this result.

Nevertheless since the very start I didn't said it's being absolute best, but close to perfection, and still requires some updates which can happen from time to time until there shall be no more of anything to further improve or readiust.

Still, know that it's really hard to watch out for more improvements on this and requires really much an eagle eye, so it might take some time for updates, and still that won't be 100% certainty to be "best". :D

**Comment by Rakso69** [comment]
You're right. Actually there is a mistake - one very little gasp in the middle-top left(4th up leftsided from armoury) spear workshop which costs this worker to travel 2-3 steps more than needed. I'll fix it tomorrow. Thanks. :)

**Comment by GRhin** [comment]
I wasnt complaining or suggesting it was wrong, I was merely wanting to understand. What are these door-change rules?

**Comment by Krarilotus** [comment]
@Rakso69 why do you use the spear production buildings? Like how much gold do they make in comparison, and why do you calculate it as more profitable?

**Comment by GRhin** [comment]
I think it's to do with number of trips to get wood. 2 wood for crossbows, vs 1 for spears, the saved time of that extra trip results in more weapons from those further workshops which makes up for the reduced value of spears.

**Comment by Rakso69** [comment]
### **Allright. This should clarify things up for everyone:**
![Best Workshops Placement2](https://user-images.githubusercontent.com/45049934/60513927-ab297280-9cd8-11e9-8a3f-bfede8496971.png)

**Comment by Rakso69** [comment]
@Krarilotus Exactly just as Grhin said. Crossbows are no longer more profitable than Spears are on **5+ workshops distance** from the stockpile.

**Comment by Krarilotus** [comment]
@Rakso69 hmmm, spears selling is worth 10 gold -4 for the wood so you get 6 gold per spear, while crossbows sell for 30 gold - 8 for the wood, so you get 22 per crossbow. I dont see the math, when it shoud be better to sell crossbows, even if you have more than 3,6666 times the pathing time? Or are the fletchers actually so slow in comparison?

**Comment by Rakso69** [comment]
They're slower on higher distance. @Krarilotus Just playtest it. I used to calculate steps efficiency only and on this distance 5+ spears walk up way less than xbows workers making spears more efficient than xbows, cuz they're being produced at a faster rate.
